src/player.js

```diff
@@ -1097,6 +1103,23 @@ export default class Player extends FakeEventTarget {
   }
 
   /**
+   * Handles the cue text removal issue, when seeking to a time without captions in IE \ edge the previous captions
+   * are not removed
+   * @returns {void}
+   * @private
+   */
+  _removeTextCuePatch(): void {
+    this._activeTextCues.forEach((textCue) => {
+      const cueEndTime = textCue._endTime;
+      const cueStartTime = textCue._startTime;
+      const currTime = this.currentTime;
+      if (currTime > cueEndTime || currTime < cueStartTime) {
```

Comment:
you need to prepare the array you send to `processCues` before calling it as it is not aware of time, just display.
Calling it like this will cause only the last cue in the array to appear